packages/repository/src/model.ts

```diff
@@ -309,7 +309,7 @@ function asObject(value: any, options?: Options): any {
 /**
  * Base class for models
  */
-export abstract class Model {
+export class Model {
```

Comment:
Can you please elaborate a bit more on why are you removing `abstract` modifier here? A link to the relevant part of TS 4.2 release notes would be great. What are the possible downsides of this change?

Reply:
https://devblogs.microsoft.com/typescript/announcing-typescript-4-2/#abstract-construct-signatures

Reply:
Thank you for the link. 

The documentation mentions "abstract constructor types", have you considered using that new type to solve the compilation problem? E.g. introduce `AbstractConstructor<T>` and use it in place that are failing to compile with `Constructor<Model>`.

Or are you trying to say that the initial choice to make Model & Entity as abstract classes was a kind of random and there are no longer reasons for treating these two base classes as abstract, thus the upgrade to TS 4.2 is a good excuse to change that? If that's the case, then I think it would be best to make this change in a standalone `feat` commit to ensure it's clearly communicated in our release notes.

Anyhow, just few ideas to consider 🤷🏻

Reply:
I choose to make the base classes `abstract` by instinct. But it does not really buy us much value. I would just take this opportunity to remove the modifier.